pkg/cloudprovider/provider/anexia/types/types.go

```diff
@@ -71,16 +81,34 @@ type RawConfig struct {
 	DiskSize int `json:"diskSize"`
 
 	Disks []RawDisk `json:"disks"`
+
+	// Deprecated, use Networks instead.
+	VlanID providerconfigtypes.ConfigVarString `json:"vlanID"`
+
+	// Configuration of the network interfaces. At least one entry with at
+	// least one Prefix is required.
+	Networks []RawNetwork `json:"networks"`
+}
+
+type NetworkAddressStatus struct {
+	ReservedIP            string    `json:"reservedIP"`
+	IPState               string    `json:"ipState"`
+	IPProvisioningExpires time.Time `json:"ipProvisioningExpires"`
+}
+
+type NetworkStatus struct {
+	// each entry belongs to a config.Networks.Prefix entry at the same index
+	Addresses []NetworkAddressStatus `json:"addresses"`
 }
 
 type ProviderStatus struct {
-	InstanceID            string         `json:"instanceID"`
-	ProvisioningID        string         `json:"provisioningID"`
-	DeprovisioningID      string         `json:"deprovisioningID"`
-	ReservedIP            string         `json:"reservedIP"`
-	IPState               string         `json:"ipState"`
-	IPProvisioningExpires time.Time      `json:"ipProvisioningExpires"`
```

Comment:
Wouldn't this be a breaking API change? We do the migration from the old VLAN identifier, so I'd assume we leave the old state filled out as well with the information of the first VLAN.

Reply:
In ANXKUBE-1131 we also specifically want to keep the old way functional for migration, so I think keeping the old status is a good idea as well

Reply:
This part of the status is not used anymore after the Machine is created, but only to allow one reconcile to reserve IPs that the following reconciles can then use to create the machine.

Keeping the status would make the code a lot more complex, for imho no real benefit (the only benefit is when updating machine-controller while a Machine is being created (IP already reserved but VM provisioning request not yet sent) - this new version cannot pick up from that and will reserve a new IP instead, the already reserved one will just timeout and be freed again by the Engine).

I did not test how this version of machine-controller behaves when reconciling an existing VM though - I would guess nothing happens, but I will check this.

> In ANXKUBE-1131 we also specifically want to keep the old way functional for migration, so I think keeping the old status is a good idea as well

The old way of configuring the MachineDeployment is still functional and translated internally in machine-controller to basically ...

``yaml
cloudProviderSpec:
  networks:
  - vlan: $cloudProviderSpec.vlanID
    prefixes:
    - "" # having an "empty-string prefix" tells the rest of the program to reserve IPs based on only the VLAN
```

Reply:
> I did not test how this version of machine-controller behaves when reconciling an existing VM though - I would guess nothing happens, but I will check this.

Oh yeah, that'd be interesting. But I think nothing big should happen at all. If you could write a test for that, it'd be greatly appreciated tho. :3

Reply:
I did test this manually by changing the machine-controller image in an existing cluster that had an existing `MachineDeployment` and `Machine`

* scaled the MachineDeployment from 1 to 2 replicas -> new Machine successfully created and Node joined the cluster, got warnings for it using the old vlanID config as expected
* scaled the MachineDeployment from 2 to 1 replicas -> the old existing Machine was chosen for deletion and that worked as expected

I don't see any benefit of putting this into an automated test, since that's only a scenario while migrating and this manual test + our knowledge about the behavior tell us "there's nothing to worry here"

Reply:
thanks mommy! I appwove this code now, you did such a great job! :pleading_face:
thank you for bapping the machine derployments awawa

Reply:
... ok I had that coming xD